sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -2308,6 +2308,7 @@ CurlConnection::CurlConnection(
     }
   }
 
+#if LIBCURL_VERSION_NUM >= 0x074D00 // 7.77.0
```

Comment:
You might consider following the pattern used above for OpenSSL 3 where we #define an internal value at the top of the file.

Reply:
I don't see how that makes the code any cleaner.  We would end up with lots of variations of USE_CURL_XXXXX for every version we end up using.  Not convinced that is any cleaner